pallets/msa/src/lib.rs

```diff
@@ -738,3 +740,75 @@ impl<T: Config> AccountProvider for Pallet<T> {
 		Ok(result.unwrap())
 	}
 }
+
+/// Check to ensure that a provider has not yet been revoked if the
```

Comment:
This deserves much more explanation, would be nice to call it out as a SignedExtension here at least.